source/fundamentals/typescript.txt

```diff
@@ -0,0 +1,97 @@
+==========
+TypeScript
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+In this guide, you can learn about the **TypeScript** features and limitiations
+of the MongoDB Node.js driver. TypeScript is a strongly typed programming
+language that compiles to JavaScript.
+
+All TypeScript features of the driver are optional. All valid JavaScript
+code written with the driver is also valid TypeScript code. 
+
+For more information, see the
+`TypeScript website <https://www.typescriptlang.org/>`__.
+
+Features
+--------
+
+If you use TypeScript, you can specify a type for some classes in the driver.
+All classes that accept a type paramater in the driver have the default type
+:node-api-4.0:`Document <interfaces/document.html>`.
```

Comment:
It isn't that useful to link to the `Document` API docs since it doesn't display the definition. Can you inline:

```typescript
interface Document {
    [key: string]: any;
}
```

This is what Document will always be defined as, the minimum constraint it enforces is that keys must be strings and `T extends Document` enforces T must be an object (as opposed to some primitive like `number`, since all things in mongodb must be objects)